service/resource_router_test.go

```diff
@@ -0,0 +1,90 @@
+package service
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/kvm-operator/service/resource/configmapv3"
+	"github.com/giantswarm/kvm-operator/service/resource/servicev2"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func Test_NewResourceRouter(t *testing.T) {
+	resourceV1 := []framework.Resource{
+		&servicev2.Resource{},
+	}
+	resourceV2 := []framework.Resource{
+		&servicev2.Resource{},
+		&configmapv3.Resource{},
+	}
+
+	testCases := []struct {
+		ctx                context.Context
+		versionedResources map[string][]framework.Resource
+		customObject       interface{}
+		expectedResources  []framework.Resource
+	}{
+		// Test 1, get resource
```

Comment:
A while ago we agreed on indexing starting at 0. Below we then do not need to increment the index by one.

Reply:
done